test/cql-pytest/cassandra_tests/distinct_query_paging_test.py

```diff
@@ -0,0 +1,179 @@
+# Cassandra source repository, as of commit 2a4cd36475de3eb47207cd88d2d472b876c6816d
+#
+# The original Apache Cassandra license:
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from cassandra_tests.porting import *
+
+from cassandra.query import BatchStatement, BatchType
+from cassandra.protocol import InvalidRequest
+
+# Migrated from cql_tests.py:TestCQL.test_select_distinct()
+def testSelectDistinct(cql, test_keyspace):
+    # Test a regular (CQL3) table.
+    with create_table(cql, test_keyspace, "(pk0 int, pk1 int, ck0 int, val int, PRIMARY KEY ((pk0, pk1), ck0))") as table:
+        for i in range(3):
+            execute(cql, table, "INSERT INTO %s (pk0, pk1, ck0, val) VALUES (?, ?, 0, 0)", i, i)
+            execute(cql, table, "INSERT INTO %s (pk0, pk1, ck0, val) VALUES (?, ?, 1, 1)", i, i)
+
+        assertRows(execute(cql, table, "SELECT DISTINCT pk0, pk1 FROM %s LIMIT 1"),
+                   row(0, 0))
+
+        assertRows(execute(cql, table, "SELECT DISTINCT pk0, pk1 FROM %s LIMIT 3"),
+                   row(0, 0),
+                   row(2, 2),
+                   row(1, 1))
+
+        # Test selection validation.
+        assertInvalidMessage(cql, table, "queries must request all the partition key columns", "SELECT DISTINCT pk0 FROM %s")
+        assertInvalidMessage(cql, table, "queries must only request partition key columns", "SELECT DISTINCT pk0, pk1, ck0 FROM %s")
+
+# Migrated from cql_tests.py:TestCQL.test_select_distinct_with_deletions()
+def testSelectDistinctWithDeletions(cql, test_keyspace):
+    with create_table(cql, test_keyspace, "(k int PRIMARY KEY, c int, v int)") as table:
+        for i in range(10):
+            execute(cql, table, "INSERT INTO %s (k, c, v) VALUES (?, ?, ?)", i, i, i)
+
+        rows = getRows(execute(cql, table, "SELECT DISTINCT k FROM %s"))
+        assert len(rows) == 10
+        key_to_delete = rows[3][0]
+
+        execute(cql, table, "DELETE FROM %s WHERE k=?", key_to_delete)
+
+        rows = getRows(execute(cql, table, "SELECT DISTINCT k FROM %s"))
+        assert len(rows) == 9
+
+        rows = getRows(execute(cql, table, "SELECT DISTINCT k FROM %s LIMIT 5"))
+        assert len(rows) == 5
+
+        rows = getRows(execute(cql, table, "SELECT DISTINCT k FROM %s"))
+        assert len(rows) == 9
+
+# Reproduces issue #10354
+@pytest.mark.xfail(reason="#10354 - we forgot to allow SELECT DISTINCT filtering on static column")
+def testSelectDistinctWithWhereClause(cql, test_keyspace):
+    with create_table(cql, test_keyspace, "(k int, a int, b int, PRIMARY KEY (k, a))") as table:
+        execute(cql, table, "CREATE INDEX ON %s (b)")
+
+        for i in range(10):
+            execute(cql, table, "INSERT INTO %s (k, a, b) VALUES (?, ?, ?)", i, i, i)
+            execute(cql, table, "INSERT INTO %s (k, a, b) VALUES (?, ?, ?)", i, i * 10, i * 10)
+
+        # In issue #10354, Scylla forgot the "and/or static columns" part.
```

Comment:
I'm not sure I fully understand the comment, when I issue such a statement, the supposedly missing part of the returned error message is present (tested with OSS 6.0):
```
cqlsh> SELECT DISTINCT pk0, pk1, ck0 FROM ms.nyh WHERE pk0 > 1 ALLOW FILTERING ;
InvalidRequest: Error from server: code=2200 [Invalid query] message="SELECT DISTINCT queries must only request partition key columns and/or static columns (not ck0)"
```
Maybe the comment needs to be rephrased.

Reply:
Okay.